packages/hierarchies-react/src/presentation-hierarchies-react/Renderers.ts

```diff
@@ -0,0 +1,98 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { GenericInstanceFilter, HierarchyNode } from "@itwin/presentation-hierarchies";
+import { InstanceKey } from "@itwin/presentation-shared";
+import { ErrorInfo, PresentationHierarchyNode } from "./TreeNode.js";
+import { SelectionChangeType } from "./UseSelectionHandler.js";
+
+/**
+ * Type definition used to render tree rendering UI component.
+ * @alpha
+ */
+export type TreeRendererProps = {
+  /**
+   * Array containing root tree nodes.
+   */
+  rootNodes: PresentationHierarchyNode[];
+  /**
+   * A function that should be called to either expand or collapse the given node.
+   */
+  expandNode: (nodeId: string, isExpanded: boolean) => void;
+  /**
+   * A function that should be called to select nodes in the tree.
+   * @param nodeIds Ids of the nodes that are selected.
+   * @param changeType Type of change that occurred for the selection.
+   */
+  selectNodes: (nodeIds: Array<string>, changeType: SelectionChangeType) => void;
+  /** Determines whether a given node is selected. */
+  isNodeSelected: (nodeId: string) => boolean;
+} & TreeErrorHandlingProps;
+
+/**
+ * Type definition used to render root error handling UI component.
+ * @alpha
+ */
+export type RootErrorRendererProps = {
+  /** Object containing root error information */
+  error: ErrorInfo;
+} & TreeErrorHandlingProps;
+
+/**
+ * Actions used to handle errors.
+ * @alpha
+ */
+interface TreeErrorHandlingProps {
```

Comment:
Are they really error handling props? I think `getHierarchyLevelDetails` is used for other cases as well

Reply:
From the tree's perspective I think it is only being used to provide solutions to errors, as it is not even passed to `NodeRenderer`. Ps. filtering action requires `getHierarchyLevelDetails`, but it is defined by consumer outside the tree.